lib/pf/role.pm

```diff
@@ -441,6 +441,7 @@ sub getRegisteredRole {
                 radius_request => $args->{radius_request},
             };
             $role = &pf::authentication::match([@sources], $params, $Actions::SET_ROLE, \$source);
+            my $value = &pf::authentication::match([@sources], $params, $Actions::SET_UNREG_DATE);
```

Comment:
You are correct @dwlfrth but it does not guarantee that the SET_UNREG_DATE action will be in the list of actions.

When you pass the action it will match the first rule and action combination.